examples/async-multisig/src/lib.rs

```diff
@@ -0,0 +1,129 @@
+#![no_std]
+
+use codec::{Decode, Encode};
+use core::str;
+use futures::{future, FutureExt};
+use gstd::{debug, msg, prelude::*, ProgramId};
+use gstd_async::msg as msg_async;
+use scale_info::TypeInfo;
+use sp_core::{
+    crypto::UncheckedFrom,
+    sr25519::{Pair as Sr25519Pair, Public, Signature},
+    Pair,
+};
+
+static mut SIGNATORIES: Vec<ProgramId> = vec![];
+static mut SIGNED_MESSAGE_PROGRAM: ProgramId = ProgramId([0u8; 32]);
+static mut THRESHOLD: usize = 0;
+
+const GAS_LIMIT: u64 = 1_000_000_000;
+
+#[derive(Debug, Encode, TypeInfo)]
+struct SignRequest {
+    message: Vec<u8>,
+}
+
+#[derive(Debug, Decode, TypeInfo)]
+pub struct SignResponse {
+    pub message: Vec<u8>,
+    pub signature: Vec<u8>,
+}
+
+gstd::metadata! {
+    title: "demo async multisig",
+    init:
+        input: Vec<u8>,
+    handle:
+        input: Vec<u8>,
+}
+
+fn hex_to_id(hex: &str) -> Result<ProgramId, ()> {
+    let hex = hex.strip_prefix("0x").unwrap_or(hex);
+
+    hex::decode(hex)
+        .map(|bytes| ProgramId::from_slice(&bytes))
+        .map_err(|_| ())
+}
+
+#[no_mangle]
+pub unsafe extern "C" fn init() {
```

Comment:
Can we make it strongly-typed?

Vec<ProgramId> would be cool

Reply:
Do you mean something like this:
```rust
#[derive(Debug, Decode, TypeInfo)]
pub struct InputArgs {
     pub threshold: usize,
     pub destination: ProgramId,
     pub signatories: Vec<ProgramId>,
 }
 
 gstd::metadata! {
    title: "demo async multisig",
    init:
        input: InputArgs,
 }
 ```
 ?

Reply:
Aye

Reply:
done